### PR TITLE
Feature/PMP-18

### PR DESCRIPTION
**Pull request by MaciejSikorski**

### CODE REVIEW

File: libs/pmp-web/user/data-access/src/lib/+state/user.effects.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { UserPartialState } from './user.reducer';
+import { UserDataService } from '../services/user-data.service';
+import { HttpErrorResponse } from '@angular/common/http';
+import { map } from 'rxjs/operators';
+
+@Injectable()
+export class UserEffects {
+  @Effect()
+  getUserStatisticsCollection$ = this.dp.fetch(fromUserActions.Types.GetUserStatisticsCollection, {
+    id: () => {},
```
Comment (wjanaszek) [inline libs/pmp-web/user/data-access/src/lib/+state/user.effects.ts]: Is this needed, when empty?

File: libs/pmp-web/user/data-access/src/lib/+state/user.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { fromUserActions } from './user.actions';
+import { HttpErrorResponse } from '@angular/common/http';
+import { UserStatistics } from '@pimp-my-pr/shared/domain';
+
+export const USER_FEATURE_KEY = 'user';
+
+export interface UserState {
+  userStatisticsCollection: UserStatistics[];
```
Comment (wjanaszek) [inline libs/pmp-web/user/data-access/src/lib/+state/user.reducer.ts]: Why no entity adapter?

File: libs/pmp-web/user/data-access/src/lib/services/user-data.service.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { map } from 'rxjs/operators';
+import { GetUserStatisticsCollectionPayload } from '../resources/payloads/get-user-statistics-collection.payload';
+import { ListUsersResponse, UserStatistics } from '@pimp-my-pr/shared/domain';
+import { urlFactory } from '@valueadd/typed-urls';
+
+@Injectable({
+  providedIn: 'root'
```
Comment (wjanaszek) [inline libs/pmp-web/user/data-access/src/lib/services/user-data.service.ts]: Shouldn't it be provided in module?

File: libs/pmp-web/user/data-access/src/lib/services/user-data.service.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+})
+export class UserDataService {
+  readonly endpoints = {
+    getUserStatisticsCollection: urlFactory('//localhost:3333/api/repository/users')
```
Comment (wjanaszek) [inline libs/pmp-web/user/data-access/src/lib/services/user-data.service.ts]: We shouldn't hardcode API url in this way

File: libs/pmp-web/user/data-access/src/lib/+state/user.reducer.ts (hunk 2 above)
Comment (stefanoww) [reply libs/pmp-web/user/data-access/src/lib/+state/user.reducer.ts]: If entity adapter will be used please add selectors from the adapter.

File: libs/pmp-web/user/users-statistics/feature/src/lib/containers/users-statistics/users-statistics.component.ts [hunk 5]
```diff
@@ -62,14 +63,20 @@ export class UsersStatisticsComponent implements OnInit {
     'link'
   ];
 
+  constructor(private chDRef: ChangeDetectorRef, private userFacade: UserFacade) {}
+
   ngOnInit(): void {
-    this.initTableConfig();
+    this.userFacade.userStatisticsCollection$.subscribe(statistics => {
```
Comment (stefanoww) [inline libs/pmp-web/user/users-statistics/feature/src/lib/containers/users-statistics/users-statistics.component.ts]: Add takeUntil

File: libs/pmp-web/user/users-statistics/feature/src/lib/containers/users-statistics/users-statistics.component.ts [hunk 6]
```diff
@@ -62,14 +63,20 @@ export class UsersStatisticsComponent implements OnInit {
     'link'
   ];
 
+  constructor(private chDRef: ChangeDetectorRef, private userFacade: UserFacade) {}
```
Comment (stefanoww) [inline libs/pmp-web/user/users-statistics/feature/src/lib/containers/users-statistics/users-statistics.component.ts]: Maybe rename to cd or cdr?

File: libs/pmp-web/user/data-access/src/lib/services/user-data.service.ts (hunk 4 above)
Comment (MaciejSikorski) [reply libs/pmp-web/user/data-access/src/lib/services/user-data.service.ts]: Yea, I know. Can it be done in another pull request to not make this one to complicated? Will it be enough when I add ToDo here?

**Pull request merged**